### ISSUE TITLE
NaN loss when training MNIST LeNet in CPU mode

### ISSUE THREAD

**Issue by wichtounet** [BVLC/caffe#5051]
### Issue summary

I just installed Caffe and tried to run the MNIST Lenet example on CPU only mode (changed solved_mode to CPU). And the loss is NAN from the first display. 

> I1202 07:38:36.619820 20673 caffe.cpp:251] Starting Optimization
> I1202 07:38:36.619825 20673 solver.cpp:279] Solving LeNet
> I1202 07:38:36.619829 20673 solver.cpp:280] Learning Rate Policy: inv
> I1202 07:38:36.620288 20673 solver.cpp:337] Iteration 0, Testing net (#0)
> I1202 07:38:40.081092 20673 solver.cpp:404]     Test net output #0: accuracy = 0.1034
> I1202 07:38:40.081132 20673 solver.cpp:404]     Test net output #1: loss = 71.5575 (* 1 = 71.5575 loss)
> I1202 07:38:40.133375 20673 solver.cpp:228] Iteration 0, loss = 68.1442
> I1202 07:38:40.133416 20673 solver.cpp:244]     Train net output #0: loss = 68.1442 (* 1 = 68.1442 loss)
> I1202 07:38:40.133440 20673 sgd_solver.cpp:106] Iteration 0, lr = 0.01
> I1202 07:38:45.040522 20673 solver.cpp:228] Iteration 100, loss = -nan
> I1202 07:38:45.040565 20673 solver.cpp:244]     Train net output #0: loss = -nan (* 1 = -nan loss)
> I1202 07:38:45.040580 20673 sgd_solver.cpp:106] Iteration 100, lr = 0.00992565
> I1202 07:38:49.861625 20673 solver.cpp:228] Iteration 200, loss = -nan
> I1202 07:38:49.861666 20673 solver.cpp:244]     Train net output #0: loss = -nan (* 1 = -nan loss)
> I1202 07:38:49.861677 20673 sgd_solver.cpp:106] Iteration 200, lr = 0.00985258

If I change it back to GPU mode, it does not happen.

What can I do to train it in CPU mode ? 
Is there a stable release of Caffe that I should try ? 

Thanks

### Steps to reproduce

(HEAD 24d2f67173db3344141dce24b1008efffbfe1c7d)

git clone https://github.com/BVLC/caffe caffe
cd caffe
make all
./data/mnist/get_mnist.sh
./examples/mnist/create_mnist.sh
./examples/mnist/train_lenet.sh

### Your system configuration
Operating system: Gentoo
Compiler: G++ 4.9.3
CUDA version (if applicable): 7.5.18
BLAS: MKL

**Comment by shelhamer** [comment]
Closing as unusual and likely the result of a faulty BLAS. CPU-mode Caffe has solver tests and I have solved this example with CPU mode by OpenBLAS and MKL in the past.
